src/core/qgscoordinatereferencesystem.h

```diff
@@ -46,38 +46,189 @@ class QgsCoordinateReferenceSystem;
 typedef void ( *CUSTOM_CRS_VALIDATION )( QgsCoordinateReferenceSystem& );
 
 /** \ingroup core
- * Class for storing a coordinate reference system (CRS)
- * \note Since QGIS 2.16 QgsCoordinateReferenceSystem objects are implicitly shared.
+ * This class represents a coordinate reference system (CRS).
+ *
+ * Coordinate reference system object defines a specific map projection, as well as transformations
+ * between different coordinate reference systems. There are various ways how a CRS can be defined:
+ * using well-known text (WKT), PROJ.4 string or combination of authority and code (e.g. EPSG:4326).
+ * QGIS comes with its internal database of coordinate reference systems (stored in SQLite) that
+ * allows lookups of CRS and seamless conversions between the various definitions.
+ *
+ * Most commonly one comes across two types of coordinate systems:
+ *
+ * 1. **Geographic coordinate systems** - based on a geodetic datum, normally with coordinates being
+ *    latitude/longitude in degrees. The most common one is World Geodetic System 84 (WGS84).
+ * 2. **Projected coordinate systems** - based on a geodetic datum with coordinates projected to a plane,
+ *    typically using meters or feet as units. Common projected coordinate systems are Universal
+ *    Transverse Mercator or Albers Equal Area.
+ *
+ * Internally QGIS uses proj4 library for all the math behind coordinate transformations, so in case
+ * of any troubles with projections it is best to examine the PROJ.4 representation within the object,
+ * as that is the representation that will be ultimately used.
+ *
+ * Methods that allow inspection of CRS instances include isValid(), authid(), description(),
+ * toWkt(), toProj4(), mapUnits() and others.
+ * Creation of CRS instances is further described in \ref crs_construct_and_copy section below.
+ * Transformations between coordinate reference systems are done using QgsCoordinateTransform class.
+ *
+ * For example, the following code will create and inspect "British national grid" CRS:
+ *
+ * ~~~{.py}
+ * crs = QgsCoordinateReferenceSystem("EPSG:27700")
+ * print "Valid:", crs.isValid()
+ * print "Description:", crs.description()
+ * print "PROJ.4:", crs.toProj4()
+ * ~~~
```

Comment:
Thanks for the hint, will try.

> Sounds nice but I wonder how it is in practice when you write a snippet that requires a layer on a postres database with a particular default value/fields/data types...?

I think the snippets would be often extracted from some Python files, and the Python files would be a part of the test suite, especially for these cases where some extra setup is necessary.